lib/src/collection/lru_map.dart

```diff
@@ -0,0 +1,286 @@
+part of quiver.collection;
+
+/**
+ * An implementation of a [Map] which has a maximum size and uses a Least
+ * Recently Used algorithm to remove items from the [Map] when the [maximumSize]
+ * is reached and new items are added.
+ */
+abstract class LruMap<K, V> implements Map<K, V> {
+  /**
+   * Creates a [LruMap] instance with the default implementation.
+   */
+  factory LruMap({int maximumSize}) = LinkedLruHashMap;
+
+  /**
+   * Maximum size of the [Map]. If [length] exceeds this value at any time, the
+   * oldest entries are removed beneath the current [length].
+   */
+  int maximumSize;
+}
+
+/**
+ * Simple implementation of a linked-list entry that contains a [key] & [value].
+ */
+class _LinkedEntry<K, V> {
+  K key;
+  V value;
+
+  _LinkedEntry<K, V> next;
+  _LinkedEntry<K, V> previous;
+
+  _LinkedEntry([this.key, this.value]);
+}
+
+/**
+ * An [Iterator]/[Iterable] combo for navigating through [_LinkedEntry].
+ */
+class _LinkedEntryIterator<K, V>
+    extends IterableBase<_LinkedEntry<K, V>>
+    implements Iterator<_LinkedEntry<K, V>> {
+  @override
+  _LinkedEntry<K, V> current;
+
+  _LinkedEntryIterator(_LinkedEntry<K, V> head) {
+    current = new _LinkedEntry<K, V>()..next = head;
+  }
+
+  @override
+  Iterator<_LinkedEntry<K, V>> get iterator => this;
```

Comment:
No, that wouldn't work either. You simply can't tie an instance of `Iterable` to an instance of an `Iterator`. `Iterable.iterator` _must_ return a new instance each time it's called. It's probably a bug that it's a getter.